roles/pulp_common/tasks/install.yml

```diff
@@ -77,6 +77,28 @@
         owner: '{{ pulp_user }}'
         group: '{{ pulp_group }}'
 
+    - name: Create media dir for Pulp
+      file:
+        path: '{{ __pulp_media_dir }}'
+        state: directory
+        owner: '{{ pulp_user }}'
+        group: '{{ pulp_group }}'
+        mode: "u+rwx,g+rwx,o+rx"
+        seuser: _default
+        serole: _default
+        setype: _default
```

Comment:
I'm fairly certain these are not needed. I'm fairly certain that the file module (like the get_url module) applies the system policy default by default, and that we install the Pulp policies (RPM or pip) as a system policy.

Although the ansible docs make it sound like they are for all but `seuser` (system policy, not user policy.).

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html

Reply:
Whenever the seuser, serole, and setype are not specifically specified the security labels look like this after creation:

```
drwxrwxr-x.  3 vagrant pulp unconfined_u:object_r:var_lib_t:s0           4096 Jan  5 21:35 media
```drwxrwxr-x.  3 vagrant pulp unconfined_u:object_r:pulpcore_var_lib_t:s0  4096 Jan  5 21:35 media


Only after the installer runs 'restorecon' later does the label change to:

```
drwxrwxr-x.  3 vagrant pulp unconfined_u:object_r:pulpcore_var_lib_t:s0  4096 Jan  5 21:35 media
```

Reply:
They are needed. Without those parameters, I am seeing the media directory created with the 'var_lib_t' context instead of 'pulpcore_var_lib_t' context. I tested on Fedora 32.

Reply:
Yes, but isn't that because the SELinux policy isn't installed or upgraded yet?

Reply:
The policy should know about it already. This change was in the policy since pulpcore-selinux 1.2.3.

https://github.com/pulp/pulpcore-selinux/commit/141feae91bece449ae8afd5321ac1a62984faa29

Reply:
But the policy doesn't get installed or upgraded until later in the installer.

install.yml calls install_pip.yml or install_packages.yml as the 2nd to last task in install.yml
https://github.com/pulp/pulp_installer/blob/master/roles/pulp_common/tasks/install.yml#L149

And it is install_pip.yml or install_packages.yml that install the policy.

Reply:
What about situations where there was already an older version of the policy installed?

This has been in the policy since 1.1.4 release that was tagged on October 12, 2020. 

https://github.com/pulp/pulpcore-selinux/releases/tag/1.1.4

Reply:
Ansible could have cached some entries. https://github.com/theforeman/pulpcore-packaging/commit/5b709f636d6da51106585316adce4f9d748308db is another example of that happening within systemd. It wouldn't surprise if this is the same thing.

Reply:
I did a test. You're right, we need to use "_default". I suspect the reason why modules like get_url worked is because the files inherited from the parent directory.